src/document-context/measure-custom-elements.ts

```diff
@@ -0,0 +1,382 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/** options passed to document.registerElement */
+interface RegisterElementOptions {
+  prototype?: {
+    createdCallback?: Function;
+
+    attachedCallback?: Function;
+    detachedCallback?: Function;
+
+    attributeChangedCallback?: Function;
+
+    _propertySetter?: Function;
+    notifyPath?: Function;
+  };
+  extends?: string;
+}
+type CustomElementV0Constructor = {
+  constructor(): HTMLElement
+};
+interface Document {
+  registerElement?(tagName: string, options: RegisterElementOptions):
+      CustomElementV0Constructor;
+}
+interface PolymerSubset {
+  (): void;
+  Async?: {
+    //
+    run(callback: Function, waitTime: number): void;
+  };
+  RenderStatus?: {
+    //
+    whenReady(cb: Function): void;
+  };
+}
+interface Window {
+  _printElementStats(): void;
+  _getElementMeasures(): ElementMeasurement[];
+  _summarizeRange(start: number, end: number): void;
+
+  customElements?: CustomElementsRegistry;
+}
+
+interface CustomElementsRegistry {
+  define(tagName: string, constructor: CustomElementV1Constructor):
```

Comment:
Yeah, the spec has it as `any` which is how I'm contributing it up for now:

https://github.com/Microsoft/TSJS-lib-generator/pull/150

Gonna leave this as is until the next typescript release when we can delete it.